src/app/c9-shell-handler.ts

```diff
@@ -21,6 +21,16 @@ class C9ShellHandler {
   private _curStatus: IShellStatus | undefined;
   private _statusCallback: StatusCallback | undefined;
 
+  constructor() {
```

Comment:
add some unit tests may be ?

Reply:
I didn't manage to mock/spy on powerMonitor.on, I have spent too much on this, I give up 🙃